### DISCUSSION TITLE
library/pwm

### DISCUSSION THREAD

**Discussion by giscus** [Air-duino/document#14] [comment]
# library/pwm

PWM，即Pulse Width Modulation（脉宽调制）是是利用微处理器的数字输出来对模拟电路进行控制的一种非常有效的技术，广泛应用在测量、通信、工控等方面。 PWM 通用 API analogWrite 此函数用于设置 PWM 输出的占空比 ulPin：要设置的 PWM 输出引脚; ulValue：占空比; AirMCU 专用 API an...

https://arduino.luatos.com/library/pwm.html

**Comment by zwguan** [comment]
请问analogWriteFrequency这个函数支持多大频率？

**Reply by wang-yupu** [reply]
我用Air001测试之后最大能有1MHz(示波器测量)，准确度在2KHz之内可以精确到1Hz

**Reply by wang-yupu** [reply]
那个准确度指的是: 频率在2000~100Hz之内(占空比随意)的实际频率与设置频率的误差在+-1Hz之内，这个可能与烧录编译设置以及芯片体质有关
